pyOCD/utility/conversion.py

```diff
@@ -38,64 +40,63 @@ def word2byte(data):
         res.append((x >> 24) & 0xff)
     return res
 
-## @brief Convert a halfword array into a byte array
-def half2byte(data):
+
+def u16leListToByteList(data):
+    """Convert a halfword array into a byte array"""
     byteData = []
     for h in data:
         byteData.extend([h & 0xff, (h >> 8) & 0xff])
     return byteData
 
-## @brief Convert a byte array into a halfword array.
-def byte2half(byteData):
+
+def byteListToU16leList(byteData):
+    """Convert a byte array into a halfword array"""
     data = []
     for i in range(0, len(byteData), 2):
-        data.append(byteData[i] | (byteData[i+1] << 8))
+        data.append(byteData[i] | (byteData[i + 1] << 8))
     return data
 
-## @brief Convert a 32-bit int to an IEEE754 float.
-def int2float(data):
-    d = struct.pack("@I", data)
-    return struct.unpack("@f", d)[0]
-
-## @brief Convert an IEEE754 float to a 32-bit int.
-def float2int(data):
-    d = struct.pack("@f", data)
-    return struct.unpack("@I", d)[0]
-
-## @brief create 8-digit hexadecimal string from 32-bit register value.
-def intToHex8(val):
-    val = hex(int(val))[2:]
-    size = len(val)
-    r = ''
-    for i in range(8-size):
-        r += '0'
-    r += str(val)
-
-    resp = ''
-    for i in range(4):
-        resp += r[8 - 2*i - 2: 8 - 2*i]
-
-    return resp
-
-## @brief Build 32-bit register value from little-endian 8-digit hexadecimal string.
-def hex8ToInt(data):
+
+def u32BEToFloat32BE(data):
+    """Convert a 32-bit int to an IEEE754 float"""
+    d = struct.pack(">I", data)
+    return struct.unpack(">f", d)[0]
```

Comment:
Ok, I could use some confirmation that this behavior is actually correct.  Previously this used `@` in the struct format string which is native byte order.  In my experience, using native is almost never actually  what is wanted in python as you don't really want to assume anything about the machine you are running on when serializing and deserializing.  Since most host PCs are big endian, I changed it to be that.